### ISSUE TITLE
Running test from test explorer using sbt build server fails in MetalsBspException

### ISSUE THREAD

**Issue by slovvik** [scalameta/metals#4189]
### Describe the bug

Given `build.sbt`

```scala
lazy val root = (project in file(".")).
configs(IntegrationTest)
  .settings(
    Defaults.itSettings,
    inThisBuild(List(
      organization := "com.example",
      scalaVersion := "2.13.6"
    )),
    name := "bsp-tests-source-sets"
  )

libraryDependencies += "org.scalatest" %% "scalatest" % "3.2.9" % Test
libraryDependencies += "org.scalatest" %% "scalatest" % "3.2.9" % IntegrationTest
```
and switched to sbt build server, when running test from text explorer metals output:
```scala
Error - 12:26:34 PM] Request workspace/executeCommand failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:484)
	at scala.concurrent.ExecutionContext$parasitic$.execute(ExecutionContext.scala:222)
	at scala.concurrent.impl.Promise$Transformation.submitWithValue(Promise.scala:429)
	at scala.concurrent.impl.Promise$DefaultPromise.submitWithValue(Promise.scala:338)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete0(Promise.scala:285)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:504)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.$anonfun$applyOrElse$4(BuildServerConnection.scala:300)
	at scala.Option.getOrElse(Option.scala:201)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:298)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:286)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:490)
	... 3 more
```

### Expected behavior

The test should be run and executed.

### Operating system

macOS

### Editor/Extension

VS Code

### Version of Metals

v0.11.17

**Comment by kpodsiad** [comment]
> and switched to sbt build server, when running test from text explorer metals output

Just to confirm, you were using bloop and then you switched to the sbt, right?

**Comment by slovvik** [comment]
yes and no :) At first I imported the build using bloop and then switched to sbt. On the second try, I removed all configuration files and folders and when opened the project with vs code I did not import it with bloop instead, I switched to sbt.

**Comment by tgodzik** [comment]
Does it work with you run the test using F5 in the specific fail containing the test? Alternatively you could switch to code lenses in the settings to see if test explorer is the culprint.

**Comment by slovvik** [comment]
It does not work in both cases. The same error occurs.

**Comment by tgodzik** [comment]
I tried to reproduce and everything seems to work ok for me. What is the sbt and JDK version that you are using? Did the code manage to compile? Does running main classes work?

**Comment by slovvik** [comment]
Sbt 1.6.2
Java 11.02

> Did the code manage to compile?

Yes

> Does running main classes work?

Yes

But I have noticed something. I will put in points on how to reproduce it.

1. Open a new window in VScode
2. Open any scala project (this in my opinion starts metals)
3. Go to metals extension
4. Create a new Scala Project from Build commands (creating a new project when no scala project is open does not work)
5. Choose scala/scala-testexample.g8 (notice that it now use sbt 1.7.1 so before import change version to 1.6.2
6. Open created project in a new window
7. Do not import the build
8. Switch build server
9. Choose sbt
10. Check if the test and the main method runs, they should run
11. Change build.sbt to one provide at the top of this issue
12. Add a new file under the src folder (it/scala/ItTest.scala)
13. Create a dummy test
14. Save the test and build.sbt
15. Import the changes 
(it gives warnings ```2022.07.27 13:06:39 WARN  sbt does not support `buildTarget/inverseSources`, unable to fetch targets owning source.
2022.07.27 13:06:39 WARN  no build target for: /Users/b.slowik/VsCodeProjects/scalatest-example3/src/it/scala/ItTest.scala```)
16. Metals do not see ItTest in the test explorer or from the file
17. Restart build server
18. Now it is able to compile the test and it is visible in metals
19. Running ItTest gives ```[Error - 1:11:30 PM] Request workspace/executeCommand failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  BuildTargetIdentifier(file:/Users/b.slowik/VsCodeProjects/scalatest-example3/#root/IntegrationTest) is not a valid build target
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:484)
	at scala.concurrent.ExecutionContext$parasitic$.execute(ExecutionContext.scala:222)
	at scala.concurrent.impl.Promise$Transformation.submitWithValue(Promise.scala:429)
	at scala.concurrent.impl.Promise$DefaultPromise.submitWithValue(Promise.scala:338)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete0(Promise.scala:285)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:504)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  BuildTargetIdentifier(file:/Users/b.slowik/VsCodeProjects/scalatest-example3/#root/IntegrationTest) is not a valid build target
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.$anonfun$applyOrElse$4(BuildServerConnection.scala:300)
	at scala.Option.getOrElse(Option.scala:201)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:298)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:286)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:490)
	... 3 more```
	
20. Close VScode, remove all the configuration files and folders
21. It should look like this 
![image](https://user-images.githubusercontent.com/12632638/181234656-3d35c22d-03d6-46ef-8a20-c1e968cdf3c5.png)

22. Open vs code and open the project (the project should open automatically)
23. Do not import the build
24. Switch build server to sbt
25. Try to run ItTest
26. Metals returns ```[Error - 1:21:29 PM] Request workspace/executeCommand failed.
  Message: Internal error.
  Code: -32603 
java.util.concurrent.CompletionException: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  
	at java.base/java.util.concurrent.CompletableFuture.encodeThrowable(CompletableFuture.java:331)
	at java.base/java.util.concurrent.CompletableFuture.completeThrowable(CompletableFuture.java:346)
	at java.base/java.util.concurrent.CompletableFuture$UniAccept.tryFire(CompletableFuture.java:704)
	at java.base/java.util.concurrent.CompletableFuture.postComplete(CompletableFuture.java:506)
	at java.base/java.util.concurrent.CompletableFuture.completeExceptionally(CompletableFuture.java:2088)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:29)
	at scala.concurrent.java8.FuturesConvertersImpl$CF.apply(FutureConvertersImpl.scala:26)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:484)
	at scala.concurrent.ExecutionContext$parasitic$.execute(ExecutionContext.scala:222)
	at scala.concurrent.impl.Promise$Transformation.submitWithValue(Promise.scala:429)
	at scala.concurrent.impl.Promise$DefaultPromise.submitWithValue(Promise.scala:338)
	at scala.concurrent.impl.Promise$DefaultPromise.tryComplete0(Promise.scala:285)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:504)
	at java.base/java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1128)
	at java.base/java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:628)
	at java.base/java.lang.Thread.run(Thread.java:834)
Caused by: scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.$anonfun$applyOrElse$4(BuildServerConnection.scala:300)
	at scala.Option.getOrElse(Option.scala:201)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:298)
	at scala.meta.internal.metals.BuildServerConnection$$anonfun$1.applyOrElse(BuildServerConnection.scala:286)
	at scala.concurrent.impl.Promise$Transformation.run(Promise.scala:490)
	... 3 more```

Before that, I did not notice the previous error.
```
scala.meta.internal.metals.MetalsBspException: BSP connection failed in the attempt to get: DebugSessionAddress.  BuildTargetIdentifier(file:/Users/b.slowik/VsCodeProjects/scalatest-example3/#root/IntegrationTest) is not a valid build target
```

**Comment by tgodzik** [comment]
This looks like an issue in https://github.com/scalacenter/scala-debug-adapter 

@adpi2 any idea what can cause it?

TLDR;

Running tests don't work with BSP sbt for itegration configuration.

**Comment by adpi2** [comment]
@tgodzik Thanks for pointing this out to me.

I confirm the integration tests are not automatically supported by the `sbt-debug-adapter` plugin. It needs to be configured manually with these settings:
```scala
lazy val root = (project in file(".")).
configs(IntegrationTest)
  .settings(
    Defaults.itSettings,
    inConfig(IntegrationTest)(DebugAdapterPlugin.testSettings), // configure the debugger for IntegrationTest 
    ...
  )
```

I don't think it is possible to add these settings automatically because we don't know if the `IntegrationTest` config was added or not in the configs of the project.

**Comment by tgodzik** [comment]
Hmm.... wouldn't it be possible to use the keys from the map here:

https://github.com/scalacenter/scala-debug-adapter/blob/main/sbt/plugin/src/main/scala/ch/epfl/scala/debugadapter/sbtplugin/DebugAdapterPlugin.scala#L78

?

**Comment by tgodzik** [comment]
Should we move the issue to scala-debug-adapter? We'll add the info about the workaround and mark it as feature request for now. Maybe someone will want to contribute?

**Comment by adpi2** [comment]
Here is the issue: https://github.com/scalacenter/scala-debug-adapter/issues/267

**Comment by tgodzik** [comment]
Thanks!
